src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/SelectExpandNode.cs

```diff
@@ -370,9 +473,8 @@ private void AddOperations(HashSet<IEdmAction> allActions, HashSet<IEdmFunction>
                 }
             }
         }
-
         // we only support paths of type 'cast/structuralOrNavPropertyOrAction' and 'structuralOrNavPropertyOrAction'.
-        internal static void ValidatePathIsSupported(ODataPath path)
+        internal static void ValidatePathIsSupportedForSelect(ODataPath path)
```

Comment:
what about nav props on a complex type nested under another complex type?

Reply:
This is existing code for testing path for $select. It should support nested complex type, but does not. Removing this check would expose issues downstream where the code assuming at most 2 segments.  Will need to be fixed as part of bigger fix.

---
In reply to: [327307928](https://github.com/OData/WebApi/pull/1738#discussion_r327307928) [](ancestors = 327307928)